src/stavenote.ts

```diff
@@ -925,86 +1005,87 @@ export class StaveNote extends StemmableNote {
    * Get the staff line and y value for the highest & lowest noteheads
    * @returns {noteHeadBounds}
    */
-  getNoteHeadBounds() {
+  getNoteHeadBounds(): StaveNoteHeadBounds {
+    if (!this.stave) throw new Vex.RERR('NoStave', 'No stave attached to this note.');
     // Top and bottom Y values for stem.
-    let yTop = null;
-    let yBottom = null;
-    let nonDisplacedX = null;
-    let displacedX = null;
+    let yTop: number | undefined;
+    let yBottom: number | undefined;
+    let nonDisplacedX: number | undefined;
+    let displacedX: number | undefined;
 
     let highestLine = this.stave.getNumLines();
     let lowestLine = 1;
-    let highestDisplacedLine = false;
-    let lowestDisplacedLine = false;
+    let highestDisplacedLine: number | undefined;
+    let lowestDisplacedLine: number | undefined;
     let highestNonDisplacedLine = highestLine;
     let lowestNonDisplacedLine = lowestLine;
 
     this.note_heads.forEach((notehead) => {
-      const line = notehead.getLine();
+      const line: number = notehead.getLine();
       const y = notehead.getY();
 
-      if (yTop === null || y < yTop) {
+      if (yTop === undefined || y < yTop) {
         yTop = y;
       }
 
-      if (yBottom === null || y > yBottom) {
+      if (yBottom === undefined || y > yBottom) {
         yBottom = y;
       }
 
-      if (displacedX === null && notehead.isDisplaced()) {
+      if (displacedX === undefined && notehead.isDisplaced()) {
         displacedX = notehead.getAbsoluteX();
       }
 
-      if (nonDisplacedX === null && !notehead.isDisplaced()) {
+      if (nonDisplacedX === undefined && !notehead.isDisplaced()) {
         nonDisplacedX = notehead.getAbsoluteX();
       }
 
       highestLine = line > highestLine ? line : highestLine;
       lowestLine = line < lowestLine ? line : lowestLine;
 
       if (notehead.isDisplaced()) {
-        highestDisplacedLine = highestDisplacedLine === false ? line : Math.max(line, highestDisplacedLine);
-        lowestDisplacedLine = lowestDisplacedLine === false ? line : Math.min(line, lowestDisplacedLine);
+        highestDisplacedLine = highestDisplacedLine === undefined ? line : Math.max(line, highestDisplacedLine);
+        lowestDisplacedLine = lowestDisplacedLine === undefined ? line : Math.min(line, lowestDisplacedLine);
       } else {
         highestNonDisplacedLine = Math.max(line, highestNonDisplacedLine);
         lowestNonDisplacedLine = Math.min(line, lowestNonDisplacedLine);
       }
     }, this);
 
     return {
-      y_top: yTop,
-      y_bottom: yBottom,
+      y_top: yTop ?? 0,
+      y_bottom: yBottom ?? 0,
```

Comment:
Do you needs these `?? 0` here? Both `yTop` and `yBottom` will be defined if we get here.

Reply:
done by initialization to numbers beyond limits 100000 & -100000.